docs/guides/marking-guide.md

```diff
@@ -0,0 +1,293 @@
+- The code follows CodeYourFuture's coding standards [[read more]](#coding-standards)
+- The code is well indented [[read more]](#well-indented-code)
+- The code uses good variable names [[read more]](#good-variable-names)
+- The code uses good function names [[read more]](#good-function-names)
+- The code uses code which is easy to read and understand [[read more]](#more-readable-code)
+- The code should not use hard coded values [[read more]](#hard-coded-values)
+
+### Clarity (/5)
+
+- The code is divided into functions of a good size. [[read more]](#small-functions)
+- Code is not repeated in several locations of the coursework. [[read more]](#non-duplicated-code)
+- The coursework does not have commented out code. [[read more]](#commented-out-code)
+- There is no dead code in the coursework [[read more]](#dead-code)
+- There is no extra or debug logging [[read more]](#debug-logging)
+
+## Explanations
+
+### Complete All Tasks
+
+Completion of all of the tasks is an essential part of the homework.
+
+If there are tests that you have to make pass, you should **not** edit them to make your code work. Doing so will mean you get marked zero for this section.
+
+### Achieves task in intended way
+
+As well as completing all of the tasks, you should be sure to complete the task in a way that we are expecting and that covers all cases.
+
+You should also always try to solve the problem in the "general" rather than in the "specific".
+
+For example, we could set the challenge
+
+> Print every element of an array to the console
+> Test 1: let array = ["A","B","C"]
+> Test 2: let array = ["1","2","3","4"]
+> Test 3: let array = ["Up", "Down"]
+
+To make this test pass you could write
+
+```js
+console.log(array[0]);
+console.log(array[1]);
+console.log(array[2]);
+console.log(array[3]);
+```
+
+However, that would not work for _every_ case that exists - only the ones that we're writing. Your aim isn't to just pass the tests, it's to write good code that solves the problem.
+
+A better solution would be
+
+```js
+array.forEach((item) => {
+  console.log(item);
+});
+```
+
+### Edge Cases
+
+Edge cases are when code does something you're not expecting and you didn't account for in your code.
+
+Some edge cases are valid data, and some are not.
+
+Some examples of valid edge case are that some names contain punctuation, or some people may only have a single name, which your code needs to be able to handle. An example of an invalid edge case is if someone doesn't enter a name at all, which your code should reject - if your code doesn't notice the problem where the user provides their name, other code you wrote may make assumptions which are incorrect.
+
+See [Falsehoods Programmers Believe About Names](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/) for some interesting examples.
+
+Good code should handle all of these cases so that they can't happen - and if they do - they don't make your program crash or stop work.
```

Comment:
```suggestion
Good code should handle all of these cases so that they can't happen - and if they do - they don't make your program crash or stop working.
```